### ISSUE TITLE
refactor TIME2STEPS into a function

### ISSUE THREAD

**Issue by behrisch** [eclipse/sumo#9429]
Macros have side effects when used with function calls. Maybe we can also rename it to seconds2sumotime or similar

**Comment by palvarezlopez** [comment]
@behrisch What do you think about the SUMOTime class? Now could be a good opportunity to implement it.

**Comment by behrisch** [comment]
No, I think a class would add too much overhead for the small advatage of type safety.

**Comment by behrisch** [comment]
Here is the renaming proposal. The goal is to avoid the usage of "step" which is very ambiguous, use "SUMOTime" whenever we refer to a long long int in milliseconds and "second" if we mean a double in seconds:

- TIME2STEPS -> seconds2SUMOTime (inline function)
- STEPS2TIME -> sumoTime2seconds (inline function)
- SIMTIME -> CURR_SECOND (stays a macro)
- SIMSTEP -> CURR_SUMOTIME (stays a macro)
- MSNet::getCurrentTimeStep -> MSNet::getCurrentSUMOTime (but use the macro whenever applicable)

I am aware that this approach uses three different capitalizations of SUMOTime and I am not happy with it. But having macros ALL_CAPS and functions always starting lower case is a good convention as well. Please improve and comment (especially @namdre :-))

**Comment by namdre** [comment]
I think SUMOTime (whatever it's capitalization) is too long for such a frequently used type. I'm also unhappy with making the succinct SIMTIME / SIMSTEP longer.
My proposal:
- rename SUMOTime to MSec
- TIME2STEPS -> sec2msec
- STEPS2TIME -> msec2sec
- SIMTIME -> SEC
- SIMSTEP -> MSEC
- MSNet::getCurrentTimeStep -> MSNet::getCurrentMSec (but use the macro whenever applicable)

Also we should do a benchmark to ensure that changing macros into inline functions does not have adverse effects.

**Comment by palvarezlopez** [comment]
@namdre but MSec can be confused with "miliseconds". I prefer "Steps"

rename SUMOTime to Step
TIME2STEPS -> sec2steps
STEPS2TIME -> steps2sec
SIMTIME -> SEC
SIMSTEP -> Step
MSNet::getCurrentTimeStep -> MSNet::getCurrentStep (but use the macro whenever applicable)

**Comment by behrisch** [comment]
To me a step is one discrete execution of the simulation loop, so one call to MSNet::simulationStep. This is not reflected in the current time at all (neither in seconds nor in the SUMOTime). That's why I want to avoid the term. But I also agree that my propsed names are long maybe too long. Furthermore it is easy to overlook the M in msec. Maybe we need a broader discussion here until someone has a good idea.

**Comment by piwagner** [comment]
I would suggest to construct it from sTime (simulation time) and s (seconds):
TIME2STEPS -> s2sTime (inline function)
STEPS2TIME -> sTime2s (inline function)
SIMTIME -> CURR_S (stays a macro)
SIMSTEP -> CURR_STIME (stays a macro)
MSNet::getCurrentSTime
